src/data/parseData.ts

```diff
@@ -58,20 +69,46 @@ function groupDataFrames(data: DataFrame[]) {
 export type ParsedData = {
   features: Feature[];
   hasTableData: boolean;
+  hasCustomTableData: boolean;
 };
 
-export function parseData(data: DataFrame[]): ParsedData {
-  const { tables, timeseries } = groupDataFrames(data);
+type chartType = 'singleTimeseries' | 'notFeatureChart' | 'featureChart';
 
+function guessChartType(tables: DataFrame[], timeseries: DataFrame[]): chartType {
   if (tables.length === 0 && timeseries.length > 0) {
-    const singleTimeseries = loadSingleTimeseries(timeseries[0].fields, timeseries[0].refId as string);
-    return {
-      features: singleTimeseries ? [singleTimeseries] : [],
-      hasTableData: false,
-    };
+    return 'singleTimeseries';
+  }
+  if (!tables[0]?.fields.some((field: { name: string }) => field.name === 'feature') && timeseries.length > 0) {
```

Comment:
@mrGnowak This `if` will be `true` even when `tables` is empty so ... it may cause a problem on line: `100: tables[0].fields` while running `loadTimeseriesWithCustomData`. Maybe our problem is out here.

Reply:
`tables[0]?.fields.some((field: { name: string }) => field.name === 'feature')` should return false, if there is no tables, or there is no SQL with name `feature`. Next, I used negation for this. Do I think right?

Reply:
You're right @mrGnowak. You fixed it 🚀